pkg/querier/distributor_queryable.go

```diff
@@ -95,34 +95,32 @@ type distributorQuerier struct {
 // Select implements storage.Querier interface.
 // The bool passed is ignored because the series is always sorted.
 func (q *distributorQuerier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
-	spanlog, ctx := spanlogger.NewWithLogger(q.ctx, q.logger, "distributorQuerier.Select")
-	defer spanlog.Finish()
+	spanLog, ctx := spanlogger.NewWithLogger(q.ctx, q.logger, "distributorQuerier.Select")
+	defer spanLog.Finish()
 
 	minT, maxT := q.mint, q.maxt
 	if sp != nil {
 		minT, maxT = sp.Start, sp.End
 	}
 
-	// If queryIngestersWithin is enabled, we do manipulate the query mint to query samples up until
-	// now - queryIngestersWithin, because older time ranges are covered by the storage. This
-	// optimization is particularly important for the blocks storage where the blocks retention in the
-	// ingesters could be way higher than queryIngestersWithin.
-	minT = int64(clampTime(q.ctx, model.Time(minT), q.queryIngestersWithin, model.Now().Add(-q.queryIngestersWithin), true, "min", "query ingesters within", spanlog))
-
-	if minT > maxT {
-		level.Debug(spanlog).Log("msg", "empty query time range after min time manipulation")
+	queryIngesters := QueryIngesters(q.queryIngestersWithin, time.Now(), q.maxt)
+	if !queryIngesters {
+		level.Debug(spanLog).Log("msg", "empty query time range after min time manipulation")
```

Comment:
[nit] Same as block store comment above - I wonder if we should use a different message here

Reply:
updated to `not querying ingesters; query time range ends before the query-ingesters-within limit`